File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        freq = 0
        ans = 0
        for i in nums:
            if(freq==0):
                ans = i
            if(ans==i):
                freq+=1
            else:
                freq-=1

        count = 0
        for i in nums:
            if(i==ans):
                count+=1
        if(count>len(nums)/2):
            return ans
        else:
            return -1

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ele=0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
        }

        for(int a:map.keySet()){
            if(map.get(a)>(nums.length/2)){
                ele = a;
            }
        }
        return ele;
    }
}